sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -0,0 +1,80 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::{BoxDynError, Error};
+use crate::postgres::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use crate::types::Type;
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct PgLTree {
+    labels: Vec<String>,
+}
+
+impl PgLTree {
+    pub fn push(&mut self, label: String) -> Result<(), Error> {
+        if label
+            .chars()
+            .all(|c| c.is_ascii_alphabetic() || c.is_ascii_digit() || c == '_')
+        {
+            self.labels.push(label);
+            Ok(())
+        } else {
+            Err(Error::InvalidLtreeLabel)
+        }
+    }
+}
+
+impl FromStr for PgLTree {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Error> {
+        Ok(Self {
+            labels: s.split('.').map(|s| s.to_owned()).collect(),
+        })
+    }
+}
+
+impl Display for PgLTree {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let mut iter = self.labels.iter();
+        if let Some(label) = iter.next() {
+            write!(f, "{}", label)?;
+            while let Some(label) = iter.next() {
+                write!(f, ".{}", label)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl Type<Postgres> for PgLTree {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::LTREE
```

Comment:
I would return `PgTypeInfo::with_name("ltree")` here and remove the entries from the `PgType` enum.

https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/postgres/type_info.rs#L215

This should solve the issue of type OID instability.

Reply:
Yes but its not using the type then

Reply:
I don't understand.

Reply:
What are you using to test this?

Reply:
Ok sorry, that was a different thing, my filter just gave 0 results

Reply:
Pushed this fix